sql/core/src/main/scala/org/apache/spark/sql/jdbc/JDBCRDD.scala

```diff
@@ -361,8 +361,14 @@ private[sql] class JDBCRDD(
           val pos = i + 1
           conversions(i) match {
             case BooleanConversion    => mutableRow.setBoolean(i, rs.getBoolean(pos))
```

Comment:
For now only Date will have the NPE problem. in #5456 when I convert the decimal type there I also apply similar approach when I call `def Decimal(x: BigDecimal)`.
I have add another test case to make sure it works for all types.